0_intro/exercises/cat/src/main.rs

```diff
@@ -1,7 +1,45 @@
+use std::env;
+use std::io;
+use std::io::prelude::*;
+use std::fs::File;
+
 pub fn cat(path: &str) -> String {
-    todo!();
+    match File::open(path) {
+        Ok(mut file) => {
+            let mut res = String::new();
+            match file.read_to_string(&mut res) {
+                Ok(_) => res,
+                Err(_) => String::from("Error while reading from file"),
+            }
+        },
+        Err(_) => String::from("Failed to open file"),
+    }
+}
```

Comment:
Very good! Good error handling even though we haven't seen it yet. If you want an improved version, you could do have `cat` return an `Option<String>` instead of returning an empty String. But that's not what was asked so it's completely fine